components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -345,16 +359,90 @@ public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumer
                         app.setTokenBindingValidationEnabled(application.isTokenBindingValidationEnabled());
                         app.setTokenRevocationWithIDPSessionTerminationEnabled(
                                 application.isTokenRevocationWithIDPSessionTerminationEnabled());
-                        app.setTokenEndpointAuthMethod(application.getTokenEndpointAuthMethod());
-                        app.setTokenEndpointAuthSignatureAlgorithm(
-                                application.getTokenEndpointAuthSignatureAlgorithm());
-                        app.setSectorIdentifierURI(application.getSectorIdentifierURI());
-                        app.setIdTokenSignatureAlgorithm(application.getIdTokenSignatureAlgorithm());
-                        app.setRequestObjectSignatureAlgorithm(application.getRequestObjectSignatureAlgorithm());
+                        String tokenEndpointAuthMethod = application.getTokenEndpointAuthMethod();
+                        if (StringUtils.isNotEmpty(tokenEndpointAuthMethod)) {
+                            if (enforceFAPIDCR) {
+                                filterAllowedFAPITokenAuthMethods(tokenEndpointAuthMethod);
+                            } else {
+                                filterTokenEndpointAuthMethods(tokenEndpointAuthMethod);
+                            }
+                            app.setTokenEndpointAuthMethod(tokenEndpointAuthMethod);
+                        }
+                        String tokenEndpointAuthSigningAlgorithm = application.getTokenEndpointAuthSignatureAlgorithm();
+                        if (StringUtils.isNotEmpty(tokenEndpointAuthSigningAlgorithm)) {
+                            if (enforceFAPIDCR) {
+                                filterAllowedFAPISignatureAlgorithms(tokenEndpointAuthSigningAlgorithm);
+                            } else {
+                                filterSignatureAlgorithms(tokenEndpointAuthSigningAlgorithm,
+                                      OAuthConstants.TOKEN_EP_SIGNATURE_ALG_CONFIGURATION);
+                            }
+                            app.setTokenEndpointAuthSignatureAlgorithm(tokenEndpointAuthSigningAlgorithm);
+                        }
+
+                        if (StringUtils.isNotEmpty(application.getSubjectType())) {
+                            OAuthConstants.SubjectType subjectType = OAuthConstants.SubjectType.fromValue(
+                                    application.getSubjectType());
+                            if (subjectType == null) {
+                                throw handleClientError(INVALID_REQUEST, "Invalid subject type");
+                            }
+                            if (OAuthConstants.SubjectType.PAIRWISE.equals(OIDCClaimUtil.getDefaultSubjectType()) ||
```

Comment:
If we add the default at the top as mentioned in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2178/files#r1371163280 we can skip `OAuthConstants.SubjectType.PAIRWISE.equals(OIDCClaimUtil.getDefaultSubjectType())`

Reply:
changed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2178/commits/fda251e87e1d07a9a8b2debc24038e717fddf9e6